klyshko-operator/controllers/tuplegenerationtask_controller.go

```diff
@@ -366,6 +371,26 @@ func (r *TupleGenerationTaskReconciler) getOrCreatePVC(ctx context.Context, key
 	return pvc, nil
 }
 
+// deletePVC deletes a PVC associated to a given task
+func (r *TupleGenerationTaskReconciler) deletePVC(ctx context.Context, key *RosterEntryKey) error {
+	logger := log.FromContext(ctx).WithValues("Task.Key", key)
+	name := types.NamespacedName{
+		Name:      pvcName(*key),
+		Namespace: key.Namespace,
+	}
+	found := &v1.PersistentVolumeClaim{}
+	err := r.Get(ctx, name, found)
+	if err == nil {
```

Comment:
It would be more idiomatic Go, if you woul remove the nesting and check for a non-nil error after each operation, i.e., Get and Delete.

Reply:
Changed.

Reply:
I think there is more to do to become idiomatic, i.e.,

```
err := r.Get(ctx, name, found)
if err == nil {
...
}
return fmt.Errorf("persistent volume claim deletion failed for task %v: %w", key, err)
```

should be 

```
err := r.Get(ctx, name, found)
if err != nil {
  return fmt.Errorf("persistent volume claim deletion failed for task %v: %w", key, err)
}
...
```

Reply:
My bad, in rebasing I realized I had a local commit that I might have forgotten to push before re-requesting the review; so obviously it had not changed for you!